micro-rdk-nmea/src/messages/pgns.rs

```diff
@@ -1,12 +1,23 @@
+use std::collections::HashMap;
+use std::marker::PhantomData;
+
+use micro_rdk::{
+    common::sensor::GenericReadingsResult,
+    google::protobuf::{value::Kind, Struct, Value},
+};
 use micro_rdk_nmea_macros::{FieldsetDerive, PgnMessageDerive};
 
+use super::message::{Message, UnparsedMessageData};
 use crate::parse_helpers::{
     enums::{Gns, GnsIntegrity, GnsMethod, RangeResidualMode, SatelliteStatus, TemperatureSource},
-    parsers::{DataCursor, FieldSet},
+    errors::NmeaParseError,
+    parsers::{DataCursor, FieldSet, NmeaMessageMetadata},
 };
 
-#[derive(PgnMessageDerive, Debug)]
+#[derive(PgnMessageDerive, Clone, Debug)]
 pub struct WaterDepth {
+    #[pgn = 128267]
+    _pgn: PhantomData<u32>,
```

Comment:
You need the PhantomData so there is something for the `pgn` macro to act on, is that right?

I'm pretty OK with it given that these are ultimately going to be generated. But, if you wanted to eliminate it, maybe it would work to just let the #pgn macro adhere to whatever was the first field and then dig it out as you iterate the fields?

Reply:
Yes that's right. I considered using whatever the first field is, but ultimately decided against it for the sake of clarity